errcheck/errcheck.go

```diff
@@ -124,28 +123,33 @@ func (e byName) Less(i, j int) bool {
 	return ei.Line < ej.Line
 }
 
+// Checker checks that you checked errors.
 type Checker struct {
```

Comment:
I think some of these fields should should probably be renamed, some have "Without" and others do not. Also I wonder if it would make sense to go for a function-based API instead of using struct fields,

Reply:
Which ones were you thinking of renaming? I imagine their zero values correspond to the defaults of the corresponding flags?

Reply:
"Blank", "Asserts" and "Tags" are kind  ambiguous in their meanings. First one would probably be better named something like "IgnoreAssignmentsToBlank" 

I can't even remember which way "Asserts" works so that probably means the name is not great :)

Reply:
"Tags" should probably be "BuildTags"

Reply:
Most of the flags in Checker struct can be separated out into a new Exclusions struct
as follows

* Checker.WithoutTests -> Exclusions.TestFiles
* Checker.WithoutGeneratedCode -> Exclusions.GeneratedFiles
* Checker.Ignore -> Exclusions.Packages (and make it []string, compute regexp inside the function)
* Checker.exclude -> Exclusions.Symbols (and make it []string, compute map inside the function)

* Checker.Blank -> Exclusions.BlankAssignments
* Checker.Asserts -> Exclusions.TypeAssertions (and invert meaning)

Then checker struct will become 

```
struct Checker {
    BuildTags []string
    Exclusions Exclusions
}
```
The absense of maps means that it can be initialized without constructor